tests/facility-settings/activity-definitions/activity-definition-form.spec.ts

```diff
@@ -0,0 +1,205 @@
+      await page.getByLabel(/description.*\*/i).fill(testData.description);
+      await page.getByLabel(/usage.*\*/i).fill(testData.usage);
+
+      await page.getByLabel(/^status$/i).click();
+      await page.getByRole("option", { name: /active/i }).click();
+
+      await page.getByRole("combobox", { name: /^category\s*\*$/i }).click();
+      await page.getByRole("option").first().click();
+      await expect(page.getByText(setup.resourceCategory.title)).toBeVisible();
+
+      await page.getByLabel(/^kind$/i).click();
+      await page.getByRole("option", { name: /service request/i }).click();
+
+      const codeCombobox = page.getByRole("combobox", { name: /^code/i });
+      await selectFromCommand(page, codeCombobox, {
+        placeholder: /search for activity codes/i,
+      });
+
+      await page
+        .getByLabel(/^derived from uri$/i)
+        .fill("https://example.com/uri");
+    });
+
+    await test.step("Additional Details", async () => {
+      const bodySite = page.getByRole("combobox", { name: /body site/i });
+      await selectFromCommand(page, bodySite, {
+        placeholder: /select body site/i,
+      });
+    });
+
+    await test.step("Requirements", async () => {
+      const specimenContainer = page
+        .getByText(/^specimen requirements$/i)
+        .locator("..");
+      const specimenTrigger = specimenContainer.getByRole("combobox").first();
+      await selectFromCommand(page, specimenTrigger, {
+        closeAfterSelect: true,
+      });
+
+      const obsContainer = page
+        .getByText(/^observation requirements$/i)
+        .locator("..");
+      const obsTrigger = obsContainer.getByRole("combobox").first();
+      await selectFromCommand(page, obsTrigger, { closeAfterSelect: true });
+
+      const chargeContainer = page
+        .getByText(/^charge item definitions$/i)
+        .locator("..");
+      const chargePicker = chargeContainer.getByRole("combobox").first();
+      await selectFromCommand(page, chargePicker, { closeAfterSelect: true });
+
+      const locationsSection = page.getByText(/^locations$/i).locator("..");
+      await locationsSection.scrollIntoViewIfNeeded();
+      const locationsTrigger = locationsSection.getByRole("combobox").first();
+      await selectFromLocationMultiSelect(page, locationsTrigger, { index: 0 });
+    });
+
+    await test.step("Diagnostic Report", async () => {
+      const diagSection = page
+        .getByText(/^diagnostic report codes$/i)
+        .locator("..");
+      const diagCombobox = diagSection.getByRole("combobox").first();
+      await selectFromCommand(page, diagCombobox, {
+        placeholder: /search for diagnostic codes/i,
+      });
+    });
+
+    await page.getByRole("button", { name: /^create$/i }).click();
+
+    await expect(
+      page.getByText(/activity definition created successfully/i),
+    ).toBeVisible();
+
+    await expect(page).toHaveURL(
+      `/facility/${setup.facility.id}/settings/activity_definitions`,
```

Comment:
I would actually navigate to the activity definition and validate what I have added is what got saved.